### PR TITLE
Complete funding information task

### PR DESCRIPTION
**Pull request by patrickjfl**
# Changes in this PR

Implements the rest of the funding information task.

NB there is a bug with the 'alternative identification' page - when 'Save and continue' is clicked with no selection, the error summary does not link to the top of the checkboxes. Any thoughts on what's going wrong very welcome :)

## Screenshots of UI changes

Screenshot 2023-11-02 at 14 20 27

Screenshot 2023-11-02 at 14 35 47

Screenshot 2023-11-02 at 14 36 01

Screenshot 2023-11-02 at 14 37 07

# Release checklist

As part of our continuous deployment strategy we must ensure that this work is
ready to be released at any point. Before merging to `main` we must first
confirm:

## Pre merge checklist

- [ ] Are any changes required to the e2e tests?
- [ ] If you've added a new route, have you added a new
  `auditEvent`? (see `server/routes/apply.ts` for examples)
- [ ] Are there environment variables or other infrastructure configuration which needs to be included in this release?
- [ ] Are there any data migrations required. Automatic or manual?
- [ ] Does this rely on changes being deployed to the CAS API?

## Post merge checklist

Once we've merged it will be auto-deployed to the dev environment.

[Find the build-and-deploy job in CircleCI](https://app.circleci.com/pipelines/github/ministryofjustice/hmpps-community-accommodation-tier-2-ui).

Should a release fail at any step, you as the author should now lead the work to
fix it as soon as possible.

### CODE REVIEW

File: server/form-pages/apply/area-and-funding/funding-information/alternativeID.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+      laRentCard,
+    }))(alternativeIDOptions)
+
+    const marriageOptions = (({ marriageCertificate, divorcePapers, dissolutionPapers }) => ({
+      marriageCertificate,
+      divorcePapers,
+      dissolutionPapers,
+    }))(alternativeIDOptions)
+
+    const financialOptions = (({ buildingSociety, councilTax, insurance, chequeBook, mortgage, savingAccount }) => ({
```
Comment (Harriethw) [inline server/form-pages/apply/area-and-funding/funding-information/alternativeID.ts:94]: wow so javascripty! Is this just an alternative to separating the `answers` by keys?, e.g. 
```
answers: {
 workAndEmploymentOptions: {
            contract: 'Employer letter/contract of employment',
            tradeUnion: 'Trade union membership card',
            invoice: 'Invoices (self-employed)',
            hmrc: 'HMRC correspondence',
            }
       }
```

**Review by Harriethw — APPROVED** [review]
this looks good, a big feature!
I think we can merge this with the understanding that there two known issues:
1. the error linking. It looks like this is an issue with the govuk checkbox component - if a divider is given as the first element it knocks the labelling/ids of the next checkboxes so the error summary can't link to them. Should be fixable by just providing the first divider outside of the checkbox items and adding custom CSS.
2. deleting conditional answers. This is the same problem we have elsewhere, if a user goes back and changes answers then it doesn't wipe the conditional questions (e.g. switching from benefits to personal savings)

I'm happy to get this merged, and then I can take on those two issues today so that you don't have to switch contexts off the Nacro views. Does that sound ok?

File: server/form-pages/apply/area-and-funding/funding-information/alternativeID.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+import { Page } from '../../../utils/decorators'
+import TaskListPage from '../../../taskListPage'
+import { convertKeyValuePairToCheckboxItems } from '../../../../utils/formUtils'
+import errorLookups from '../../../../i18n/en/errors.json'
+import { nameOrPlaceholderCopy } from '../../../../utils/utils'
+import { getQuestions } from '../../../utils/questions'
+
+const applicationQuestions = getQuestions('')['funding-information']['alternative-identification']
+
+const alternativeIDOptions = applicationQuestions.alternativeIDDocuments.answers
```
Comment (Harriethw) [inline server/form-pages/apply/area-and-funding/funding-information/alternativeID.ts:12]: is there a reason why sometimes there are variables declared outside of the class? I don't have a strong opinion either way, but just curious as elsewhere I've just been assigning things like `questions` when they are declared in the class https://github.com/ministryofjustice/hmpps-community-accommodation-tier-2-ui/blob/314d499ae6d0c830fb46afd28e3b5aeb23f8764b/server/form-pages/apply/offence-and-licence-information/offending-history/anyPreviousConvictions.ts

File: server/form-pages/apply/area-and-funding/funding-information/alternativeID.ts (hunk 1 above)
Comment (patrickjfl) [reply server/form-pages/apply/area-and-funding/funding-information/alternativeID.ts:94]: haha I know! yes it is, I thought that given `workAndEmploymentOptions` are not really a concept in our data, this approach was preferable to modifying the `questions` data structure and using a custom `response` method.

File: server/form-pages/apply/area-and-funding/funding-information/alternativeID.ts (hunk 2 above)
Comment (patrickjfl) [reply server/form-pages/apply/area-and-funding/funding-information/alternativeID.ts:12]: I think I've been doing it in cases where the body type definition relies on the answer keys (as on line 15 here). I don't think we can define types within the class...?

**Comment by patrickjfl** [comment]
> this looks good, a big feature! I think we can merge this with the understanding that there two known issues:
> 
>     1. the error linking. It looks like this is an issue with the govuk checkbox component - if a divider is given as the first element it knocks the labelling/ids of the next checkboxes so the error summary can't link to them. Should be fixable by just providing the first divider outside of the checkbox items and adding custom CSS.
> 
>     2. deleting conditional answers. This is the same problem we have elsewhere, if a user goes back and changes answers then it doesn't wipe the conditional questions (e.g. switching from benefits to personal savings)
> 
> 
> I'm happy to get this merged, and then I can take on those two issues today so that you don't have to switch contexts off the Nacro views. Does that sound ok?

That sounds good! Thanks for working out the error message issue :)

**Pull request merged**